### PR TITLE
[2] docs: previewer explanation

### LINKED ISSUE #3973: documentation: previewer
** Read before you start ** https://www.divio.com/blog/documentation/ (Explanation section)
Check the parent task: https://github.com/inveniosoftware/invenio/issues/3882 and comment for structure of the documentation, and write documentation for sections listed below:

	- Previewing files
	- IIIF image previewer

### PR DESCRIPTION
**Pull request by equadon**
Closes #3973

### CODE REVIEW

File: docs/managing-files/previewer.rst [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+.. _preview-with-iiif:
+
+Preview images using IIIF
+-------------------------
+Invenio-IIIF implements the `IIIF API <https://iiif.io/>`_ which is a standard
+for describing and delivering images over the web. `Invenio-IIIF <https://invenio-iiif.readthedocs.io/en/latest/>`_
+allows you to generate thumbnails, resize and zoom images as well as previewing
+images using Invenio-Previewer.
+
+For Invenio-IIIF to work the ``iiif_image`` previewer needs to be activated
```
Comment (kprzerwa) [inline docs/managing-files/previewer.rst]:
```suggestion
To activate the IIIF Image previewer you need to add ``iiif_image`` into the list of supported previewers ``PREVIEWER_PREFERENCE``, e.g.:
```

File: docs/managing-files/previewer.rst [hunk 2]
```diff
@@ -0,0 +1,92 @@
...
+        'xml_prismjs',
+        'pdfjs',
+        'zip',
+    ]
+
+The IIIF API is available by default on ``/api/iiif`` and to generate
+valid image URLs, the utilify function `invenio_iiif.utils.ui_iiif_image_url <https://invenio-iiif.readthedocs.io/en/latest/api.html#invenio_iiif.utils.ui_iiif_image_url>`_
+can be used. Given that you have an image object ``img_obj`` belonging to a bucket
+``bucket`` (see :ref:`managing-files`), an image URL can be generated for that
+image like this:
```
Comment (kprzerwa) [inline docs/managing-files/previewer.rst]:
```suggestion
image as in the example below:
```

File: docs/managing-files/previewer.rst [hunk 3]
```diff
@@ -0,0 +1,92 @@
...
+    from invenio_iiif.utils import ui_iiif_image_url
+    image_url = ui_iiif_image_url(
+          obj=img_obj, version='v2', region='full', size='full', rotation=0,
+          quality='default', image_format='png')
+
+The generated image URL will look similar to this:
+
+``/iiif/<bucket_id>:<version_id>:img.png/full/full/0/default.png``
+
+For a more complete guide on how to use Invenio-IIIF see the
```
Comment (kprzerwa) [inline docs/managing-files/previewer.rst]:
```suggestion
For a more complete guide on how to use Invenio-IIIF and detailed documentation, see the
```

File: docs/managing-files/previewer.rst [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+Previewing files
+================
+Invenio has support for previewing many of the most popular file formats
+including PDF, ZIP, Markdown, images and Jupyter Notebooks. In this section we
+will discuss how previewing files work in Invenio and which options are
+available.
+
+`Invenio-Previewer <https://invenio-previewer.readthedocs.io/en/latest/>`_
+allows you to register previewers based on the file extension. To preview
+files, ensure that a preview endpoint have been configured in your
```
Comment (roksys) [inline docs/managing-files/previewer.rst]: I might be wrong, but I think it's more common to use`has been` with the third person?

File: docs/managing-files/previewer.rst [hunk 5]
```diff
@@ -0,0 +1,95 @@
...
+    RECORDS_UI_ENDPOINTS=dict(
+        recid_previewer=dict(
+            pid_type='recid',
+            route='/records/<pid_value>/preview/<filename>',
+            view_imp='invenio_previewer.views:preview',
+            record_class='invenio_records_files.api:Record',
+        ),
+    )
+
+Upload a file (see :ref:`managing-files`) of a type supported by the previewer,
```
Comment (roksys) [inline docs/managing-files/previewer.rst]: Once, there is a full `managing-files` section, we should update the reference to point to the exact file upload step.

**Comment by ntarocco** [comment]
Will take all comments into account.

**Pull request merged**